### PR TITLE
utilise automatic registration of compiler passes

### PR DESCRIPTION
**Pull request by Guite**
| Q                 | A
| ----------------- | ---
| Bug fix?          | no
| New feature?      | no
| BC breaks?        | no
| Deprecations?     | no
| Tests pass?       | yes
| Fixed tickets     | #2790
| Refs tickets      | -
| License           | MIT
| Doc PR            | -
| Changelog updated | no

### CODE REVIEW

**Comment by Guite** [comment]
This apparently causes problems during the CLI install process (see Travis).: `The service "zikula.doctrine.schema_tool" has a dependency on a non-existent service "doctrine.entitymanager".`
The `doctrine.entitymanager` alias is created by the DoctrinePass class here: https://github.com/zikula/core/blob/1.4/src/lib/Zikula/Bundle/CoreBundle/DependencyInjection/Compiler/DoctrinePass.php#L14
It is not recognised automatically...

**Comment by Guite** [comment]
Probably composer tries to execute stuff before the container has been compiled.

**Comment by craigh** [comment]
why is this advantageous?

**Comment by Guite** [comment]
Sorry I don't understand your comment.

**Comment by craigh** [comment]
why is it better to "utilise automatic registration of compiler passes"? why not the original way?

**Comment by craigh** [comment]
I guess my question seems dumb because I am the one that submitted the original ticket (#2790). I had forgotten I had done so :blush: 

Even so, do you plan to try to get this to work or should we close the PR?

**Comment by Guite** [comment]
I tend to close both the ticket as well as the PR, since it seems we cannot easily fix it (unless somebody has an idea about it)...

**Comment by craigh** [comment]
I think we just need to change the dependency service definition from `doctrine.entitymanager` to the original (non-aliased). can't think of what it is right now. `php app/console debug:container` will give you all the service names. you can look it up.

**Comment by craigh** [comment]
@Guite - rebase this branch/PR please

**Comment by Guite** [comment]
After I changed the service names in several other places the travis build now works.

**Comment by craigh** [comment]
rebase is not the same as merge.

**Comment by craigh** [comment]
locally, please:

  1) rebase to HEAD of `zikula:1.4` branch
  2) soft reset to last core commit
  3) commit all the changes with one commit. Use CONTRIBUTING header as the commit message
  4) force push to `Guite:compilerpassinterface`

If you need assistance with this, I will be happy to walk you through it. I will be available about 20:00 your time.

**Comment by Guite** [comment]
Seems I've coped with it.

**Comment by craigh** [comment]
> Seems I've coped with it.

much better! and thank you. Have you tested locally?

File: src/lib/Zikula/Bundle/CoreBundle/DependencyInjection/Compiler/DoctrinePass.php [hunk 1]
```diff
@@ -1,4 +1,15 @@
 <?php
+/**
+ * Copyright Zikula Foundation 2015 - Zikula Application Framework
```
Comment (craigh) [inline src/lib/Zikula/Bundle/CoreBundle/DependencyInjection/Compiler/DoctrinePass.php]: all the headers say 2015. It is *2016* now :smile:

Comment (craigh) [inline src/lib/Zikula/Bundle/CoreBundle/DependencyInjection/Compiler/DoctrinePass.php]: I don't care about this really. eventually we need to replace the header entirely.

**Comment by cmfcmf** [comment]
:+1:

**Comment by craigh** [comment]
@Guite - some quick testing on this branch shows it doesn't work.

screen shot 2016-03-23 at 8 53 00 pm

The rest of the trace is too long to post. I'm not sure what the problem is but it seems to only affect this branch. Please take a look and correct.

**Comment by Guite** [comment]
Can not reproduce locally. Do you remember which specific pages you called?

**Comment by craigh** [comment]
> Do you remember which specific pages you called?

any. I just checked out the branch, manually cleared by cache and every page resulted in that error. :(

**Comment by Guite** [comment]
I did a fresh install using this branch and it worked - at least the pages I called, for example Extensions admin, Mailer admin, admin category showing modules, and some others.

**Comment by craigh** [comment]
I will try again

**Comment by Guite** [comment]
Just the be sure about theory: there is no difference between doing a new install and clearing the cache in an existing one, right? Symfony just recompiles the container which should include the compiler passes anyway.

**Comment by craigh** [comment]
> there is no difference between doing a new install and clearing the cache in an existing one, right?

not entirely correct. There are other artifacts which which can affect the situation. A *closer* approximation to a new install would be to clear all caches (delete the entire directories), clear `/web` (save `.htaccess`), `/vendors`, `/logs`, then removing `app/config/custom_parameters.yml`, `app/config/dynamic/generated.yml`, `config/personal_config.php` and all tables in the DB.

then `composer install`
then install core

**Comment by craigh** [comment]
I completely wiped my installation and started from scratch (did what I said above) and this branch **worked OK** so I guess it is clean to merge. I'm guessing it will require something like that in all dev environments though when merged. Although I don't know why.

**Comment by craigh** [comment]
well, the link collector isn't working anymore in some cases. very confusing.

**Comment by craigh** [comment]
I reset this commit and force pushed to eliminate this merge entirely from the history. It broke the compiler passes, so there is obviously something not working. I shouldn't have merged it so quickly.

**Pull request merged**